app/jobs/email_privacy_policy_update_job.rb

```diff
@@ -0,0 +1,55 @@
+class EmailPrivacyPolicyUpdateJob < ApplicationJob
+  def perform(petition, limit: 1000)
+    terminating = false
+
+    worker = trap("TERM") do
+      terminating = true
+      worker.call
+    end
+
+    Appsignal.without_instrumentation do
+      signatures(petition, limit).each do |signature|
+        send_email(signature)
+        update_records(signature)
+
+        if terminating
+          reschedule_job(petition, limit, time: 5.minutes.from_now)
+          return true
+        end
+      end
+    end
+
+    if signatures(petition, limit).exists?
+      reschedule_job(petition, limit)
+    end
+
+  ensure
+    trap("TERM", worker)
+  end
+
+  private
+
+  def signatures(petition, limit)
+    petition
+      .signatures
+      .where(privacy_policy_update_sent: nil)
+      .batch(limit: limit)
+  end
+
+  def send_email(signature)
+    PetitionMailer
+      .with(signature: signature)
+      .privacy_policy_update_email
+      .deliver_now
+  end
+
+  def update_records(signature)
+    Signature
+      .where(email: signature.email)
+      .update_all(privacy_policy_update_sent: true)
```

Comment:
Avoid sending more than one email per signatory by setting `privacy_policy_update_sent` to `true` for all `signatures` with the same `email`.